### PR TITLE
OSIDB-2740: Rename Flaw summary and description

### PR DESCRIPTION
**Pull request by MrMarble**
It was identified that the naming OSIDB introduced in its ancient times is not appropriate and people are constantly confused by it.

We should rename `Flaw.description` which corresponds to the initial Bugzilla flaw bug comment to `comment_zero`. Additionally, we should rename `Flaw.summary` to `cve_description` - we **CANNOT** use just description as it would make the transition very complicated by conflicting with the current name of the initial Bugzilla flaw comment.

To keep the value aligned this will also require to rename `Flaw.requires_summary` to `Flaw.requires_cve_description`.

Closes OSIDB-2740

> This task was bigger than I initially though, should have done it in two separate pull requests... also my local environment went crazy and had to rely on CI for tests, sorry for the rebase mess

### CODE REVIEW

File: apps/bbsync/query.py [hunk 1]
```diff
@@ -136,11 +136,11 @@ def generate_unconditional(self):
         """
         generate query attributes not requiring conditional processing
         """
-        self._query["cf_release_notes"] = self.flaw.summary
+        self._query["cf_release_notes"] = self.flaw.cve_description
         self._query["severity"] = self.IMPACT_TO_SEVERITY_PRIORITY[self.flaw.impact]
         self._query["priority"] = self.IMPACT_TO_SEVERITY_PRIORITY[self.flaw.impact]
 
-    def generate_summary(self):
+    def generate_cve_description(self):
```
Comment (JakubFrejlach) [inline apps/bbsync/query.py:143]: I am thinking about whether this function should actually be renamed or not, as basically this serves merely for generating the data for Bugzilla and in Bugzilla, this field will still be named `summary` :thinking: Maybe just updating the doc text of that function and clearly state the `cve_description` <-> `summary` info?

File: apps/bbsync/query.py [hunk 2]
```diff
@@ -181,12 +181,12 @@ def generate_summary(self):
         component = self.flaw.component + ": " if self.flaw.component else ""
         self._query["summary"] = embargoed + cve_ids + component + self.flaw.title
 
-    def generate_description(self):
+    def generate_comment_zero(self):
```
Comment (JakubFrejlach) [inline apps/bbsync/query.py:184]: Same as above.

File: collectors/bzimport/fixups.py [hunk 3]
```diff
@@ -171,13 +171,13 @@ def fix_title(self) -> None:
         we strip the summary prefixes without checking them
         and then try to get and store the component
         """
-        if "summary" not in self.flaw_json:
+        if "cve_description" not in self.flaw_json:
```
Comment (JakubFrejlach) [inline collectors/bzimport/fixups.py:174]: I think `flaw_json` contains the Flaw data obtained from Bugzilla and thus there will be still `summary`.

File: collectors/bzimport/fixups.py [hunk 4]
```diff
@@ -171,13 +171,13 @@ def fix_title(self) -> None:
...
             self.errors.append("no summary")
             # here we do not have to default as the
             # non-empty value is enforced by the model
             return
 
-        title = self.flaw_json["summary"]
+        title = self.flaw_json["cve_description"]
```
Comment (JakubFrejlach) [inline collectors/bzimport/fixups.py:180]: Same as above.

File: collectors/bzimport/fixups.py [hunk 5]
```diff
@@ -211,8 +211,8 @@ def fix_title(self) -> None:
 
     def fix_description(self) -> None:
         """description fixup"""
-        if "description" in self.flaw_json:
-            self.flaw_obj.description = self.flaw_json["description"]
+        if "comment_zero" in self.flaw_json:
+            self.flaw_obj.comment_zero = self.flaw_json["comment_zero"]
```
Comment (JakubFrejlach) [inline collectors/bzimport/fixups.py:215]: Same as above.

File: collectors/bzimport/tests/test_convertors.py [hunk 6]
```diff
@@ -685,14 +685,14 @@ def get_flaw_bug(cls):
             "component": "vulnerability",
             "creation_time": "2000-01-01T12:12:12Z",
             "depends_on": [],
-            "description": "text",
```
Comment (JakubFrejlach) [inline collectors/bzimport/tests/test_convertors.py:688]: This data are supposed to be from Bugzilla, `description` there remains unchanged.

File: collectors/bzimport/tests/test_convertors.py [hunk 7]
```diff
@@ -685,14 +685,14 @@ def get_flaw_bug(cls):
...
+            "comment_zero": "text",
             "fixed_in": None,
             "groups": [],
             "id": "123",
             "last_change_time": "2001-01-01T12:12:12Z",
             "status": "CLOSED",
             "resolution": "",
-            "summary": "EMBARGOED TRIAGE CVE-2000-1234 foo: ACL bypass with Authorization: 0000 HTTP header",
+            "cve_description": "EMBARGOED TRIAGE CVE-2000-1234 foo: ACL bypass with Authorization: 0000 HTTP header",
```
Comment (JakubFrejlach) [inline collectors/bzimport/tests/test_convertors.py:695]: This data are supposed to be from Bugzilla, `summary` there remains unchanged.

File: collectors/bzimport/tests/test_fixups.py [hunk 8]
```diff
@@ -24,7 +24,7 @@ def test_affect_ps_module_fixup(self):
 
 class TestFlawFixer:
     @pytest.mark.parametrize(
-        "summary,title,component,components",
+        "cve_description,title,component,components",
```
Comment (JakubFrejlach) [inline collectors/bzimport/tests/test_fixups.py:27]: This should also remain the same as this is supposed to be summary coming from Bugzilla. This applies for the whole test below.

File: osidb/tests/factories.py [hunk 9]
```diff
@@ -98,7 +98,7 @@ class Params:
         "random_element", elements=list(set(Impact) - {Impact.NOVALUE})
     )
     components = factory.List([factory.Faker("word") for _ in range(3)])
-    description = factory.LazyAttribute(lambda c: f"Description for {c.cve_id}")
+    comment_zero = factory.LazyAttribute(lambda c: f"Description for {c.cve_id}")
```
Comment (JakubFrejlach) [inline osidb/tests/factories.py:101]: Maybe also change the actual text in lambda function to `Comment zero  for ...`

File: osidb/tests/factories.py [hunk 10]
```diff
@@ -521,7 +521,7 @@ def content(self):
                 }
             ],
             "cwe_id": "CWE-110",
-            "description": "some description",
+            "comment_zero": "some description",
```
Comment (JakubFrejlach) [inline osidb/tests/factories.py:524]: Text change for consistency.

File: osidb/tests/test_flaw.py [hunk 11]
```diff
@@ -1075,34 +1077,34 @@ def test_validate_impact_and_summary(self, impact, summary, should_alert, alert)
 
         if should_alert:
             assert flaw.alerts.count() == 1
-            assert flaw.alerts.filter(name="impact_without_summary").exists()
+            assert flaw.alerts.filter(name=alert).exists()
```
Comment (JakubFrejlach) [inline osidb/tests/test_flaw.py:1080]: I wonder how this worked before? :smile:

Comment (MrMarble) [reply osidb/tests/test_flaw.py:1080]: The only alert used on the test was `impact_without_summary` that's why it worked 😆, but the `alert` variable was declared

File: apps/workflows/tests/test_models.py [hunk 12]
```diff
@@ -424,7 +424,7 @@ def test_classify(self):
         flaw = Flaw()
         assert_state_equals(workflow.classify(flaw), state_new)
 
-        flaw.description = "valid description"
+        flaw.comment_zero = "valid description"
```
Comment (jobselko) [inline apps/workflows/tests/test_models.py:427]: `flaw.comment_zero = "valid comment_zero"`

File: apps/workflows/tests/test_models.py [hunk 13]
```diff
@@ -548,14 +548,14 @@ def test_classify_complete(self):
...
-        flaw.description = ""
+        flaw.comment_zero = ""
         flaw.title = ""
 
         classified_workflow, classified_state = workflow_framework.classify(flaw)
         assert_workflow_equals(classified_workflow, workflow_main)
         assert_state_equals(classified_state, state_new)
 
-        flaw.description = "valid description"
+        flaw.comment_zero = "valid description"
```
Comment (jobselko) [inline apps/workflows/tests/test_models.py:558]: `flaw.comment_zero = "valid comment_zero"`

File: apps/workflows/tests/test_models.py [hunk 14]
```diff
@@ -725,14 +725,14 @@ def test_promote(self):
...
-        with pytest.raises(MissingRequirementsException, match="has summary"):
+        with pytest.raises(MissingRequirementsException, match="has cve_description"):
             flaw.promote()
         assert (
             flaw.classification["state"]
             == WorkflowModel.WorkflowState.SECONDARY_ASSESSMENT
         )
 
-        flaw.summary = "valid summary"
+        flaw.cve_description = "valid summary"
```
Comment (jobselko) [inline apps/workflows/tests/test_models.py:735]: `flaw.cve_description = "valid cve_description"`

File: collectors/bzimport/convertors.py [hunk 15]
```diff
@@ -654,7 +654,7 @@ def get_meta_attr(self, cve_id):
         meta_attr["alias"] = self.alias
         meta_attr["depends_on"] = self.depends_on
         meta_attr["related_cves"] = [c for c in self.cve_ids if c != cve_id]
-        meta_attr["bz_summary"] = self.flaw_bug["summary"]
+        meta_attr["bz_summary"] = self.flaw_bug["cve_description"]
```
Comment (jobselko) [inline collectors/bzimport/convertors.py:657]: `self.flaw_bug` contains data in Bugzilla format, so `summary` should stay here (Bugzilla cannot recognize `cve_description`)

File: osidb/tests/factories.py [hunk 16]
```diff
@@ -126,24 +126,24 @@ class Params:
             # Flaw.FlawNistCvssValidation.REJECTED,
         ],
     )
-    summary = factory.LazyAttribute(
+    cve_description = factory.LazyAttribute(
         lambda f: "I am a spooky CVE"
         if f.is_mi
-        else ("random summary" if f.impact_requiring_summary else f.fallback)
+        else ("random cve_description" if f.impact_requiring_summary else f.fallback)
```
Comment (jobselko) [inline osidb/tests/factories.py:132]: `f.impact_requiring_summary` should be renamed to `f.impact_requiring_cve_description`. Similarly for `not_mi_with_summary` and  `not_mi_without_summary`

File: osidb/tests/test_search.py [hunk 17]
```diff
@@ -107,28 +107,28 @@ def test_search_flaws_on_update(
...
         assert body["count"] == 0
 
-        response = auth_client().get(f"{test_api_uri}/flaws?search=summary")
+        response = auth_client().get(f"{test_api_uri}/flaws?search=cve_description")
         assert response.status_code == 200
         body = response.json()
         assert body["count"] == 1
 
-        flaw.summary = "NOMORESUMMARY"
+        flaw.cve_description = "NOMORESUMMARY"
```
Comment (jobselko) [inline osidb/tests/test_search.py:128]: ` flaw.cve_description = "NOMORECVEDESCRIPTION"`

**Review by jobselko — CHANGES_REQUESTED** [review]

**Comment by MrMarble** [comment]
Related to https://github.com/RedHatProductSecurity/osim/pull/242 and https://github.com/RedHatProductSecurity/osim/pull/240

File: osidb/models.py [hunk 18]
```diff
@@ -1506,6 +1511,7 @@ def _create_flaw(self, *args, **kwargs) -> Flaw:
 
         # Flaw model has to be created first
         model, data = [i for i in main_model.items()][0]
+        print(data)
```
Comment (osoukup) [inline osidb/models.py:1514]: This seems like forgotten temporary print :smiley:

File: osidb/tests/factories.py [hunk 19]
```diff
@@ -492,19 +504,23 @@ class Params:
...
         )
         url = factory.LazyAttribute(
-            lambda f: f"https://nvd.nist.gov/vuln/detail/{f.ext_id}"
-            if f.source == Snippet.Source.NVD
-            else f"https://osv.dev/vulnerability/{f.ext_id}"
+            lambda f: (
+                f"https://nvd.nist.gov/vuln/detail/{f.ext_id}"
+                if f.source == Snippet.Source.NVD
+                else f"https://osv.dev/vulnerability/{f.ext_id}"
+            )
```
Comment (osoukup) [inline osidb/tests/factories.py:511]: I am thinking what is different here, maybe a different version of **black**?

**Review by osoukup — CHANGES_REQUESTED** [review]
One minor change request, otherwise LGTM. Quite a lot of work at the end! Thank you for addressing all of it!

File: osidb/models.py (hunk 18 above)
Comment (MrMarble) [reply osidb/models.py:1514]: oops yes 😅

File: osidb/tests/factories.py (hunk 19 above)
Comment (MrMarble) [reply osidb/tests/factories.py]: Yes, for some reason the `make format` command was throwing an `invalid ssl certificate` error when trying to install black dependencies so I just ran my local black instance (with the same arguments), I've fixed the error of `make format` and is working now

Comment (osoukup) [reply osidb/tests/factories.py]: I do not have any serious objections against it until the **black** here in the pipeline is satisfied.

**Review by jobselko — CHANGES_REQUESTED** [review]
I am sorry to request other changes, but I missed a lot of things during the previous review. I hope the collector part is not too chaotic. If anything is unclear, just ping me! :)

File: osidb/migrations/0142_auto_20240605_0822.py [hunk 20]
```diff
@@ -0,0 +1,28 @@
+# Generated by Django 3.2.25 on 2024-06-05 08:22
```
Comment (osoukup) [inline osidb/migrations/0142_auto_20240605_0822.py:1]: The migration file could have a human-readable name

**Review by jobselko — APPROVED** [review]
LGTM, great job!

File: apps/bbsync/tests/test_integration.py
Comment (jobselko) [inline apps/bbsync/tests/test_integration.py:1]: Regarding the failing `test_flaw_draft_create`: the easiest way to fix it is to manually change all occurrences of `"some description"` to `"some comment_zero"` in three related cassettes. (But running `record-rewrite` is also possible).

File: collectors/nvd/collectors.py [hunk 21]
```diff
@@ -111,7 +111,7 @@ def get_cwes(data: CVE) -> Union[str, None]:
             else:
                 return None
 
-        def get_description(descriptions: list) -> Union[str, None]:
+        def get_comment_zero(descriptions: list) -> Union[str, None]:
             """
             Return English description from `descriptions`.
```
[the remark this answers is not in the document]
Comment (jobselko) [reply collectors/nvd/collectors.py:116]: I overlooked this one, thanks!

**Pull request merged**